backend/src/graphql_api/plt.rs

```diff
@@ -471,49 +471,146 @@ impl PLTToken {
+        #[graphql(desc = "Returns the elements in the list that come before the specified cursor.")]
+        before: Option<String>,
+    ) -> ApiResult<
+        connection::Connection<NestedCursor<DescendingI64, DescendingI64>, PltAccountAmount>,
+    > {
         let pool = get_pool(ctx)?;
-        let token_id: TokenId = token_id.to_string();
-        Ok(PLTAccountAmount::query_by_token_id(pool, token_id).await?)
+        let token_id = TokenId::from_str(token_id.as_ref())
+            .map_err(|e| ApiError::InvalidIdFormat(format!("Failed to parse token ID: {}", e)))?;
+        let config = get_config(ctx)?;
+        // Nested cursor: order by amount DESC, then account_index DESC for
+        // deterministic pagination This ensures accounts with same token
+        // amounts have a strict ordering
+        type Cursor = NestedCursor<DescendingI64, DescendingI64>;
+        let query = ConnectionQuery::<Cursor>::new(
+            first,
+            after,
+            last,
+            before,
+            config.plt_account_amount_connection_limit,
+        )?;
+
+        let mut row_stream = sqlx::query_as!(
+            PltAccountAmount,
+            "SELECT 
+                pa.account_index,
+                pa.token_index,
+                pa.amount,
+                pa.decimal
+            FROM plt_accounts pa
+            JOIN plt_tokens pt ON pa.token_index = pt.index
+            WHERE pt.token_id = $1
+              AND (
+                  (COALESCE(pa.amount, 0) > $3 AND COALESCE(pa.amount, 0) < $2)
+                  OR ($2 != $3 AND (
+                      (COALESCE(pa.amount, 0) = $2 AND pa.account_index < $5)
+                      OR (COALESCE(pa.amount, 0) = $3 AND pa.account_index > $4)
+                  ))
+                  OR ($2 = $3 AND COALESCE(pa.amount, 0) = $2 AND pa.account_index < $5 AND \
+             pa.account_index > $4)
+              )
+            ORDER BY 
+                CASE WHEN $7 THEN COALESCE(pa.amount, 0) END ASC,
+                CASE WHEN $7 THEN pa.account_index END ASC,
+                CASE WHEN NOT $7 THEN COALESCE(pa.amount, 0) END DESC,
+                CASE WHEN NOT $7 THEN pa.account_index END DESC
+            LIMIT $6",
+            token_id.to_string(),
+            BigDecimal::from(i64::from(query.from.outer)), // $2 - amount upper bound
+            BigDecimal::from(i64::from(query.to.outer)),   // $3 - amount lower bound
+            i64::from(query.from.inner),                   // $4 - account_index upper bound
+            i64::from(query.to.inner),                     // $5 - account_index lower bound
+            query.limit,
+            query.is_last,
+        )
+        .fetch(pool);
+        let mut connection = connection::Connection::new(false, false);
+        while let Some(tx) = row_stream.try_next().await? {
+            let amount_val = tx.amount.as_ref().and_then(|a| a.to_i64()).unwrap_or(0);
```

Comment:
Instead of using a default value 0, suggestion would be to return an internal error so we catch that something is off. I would expect that the logic of the index stores an i64 into the database, so we should be able to parse that out.